testproject/Assets/Tests/Runtime/MultiprocessRuntime/ExecuteStepInContextTests.cs

```diff
@@ -142,31 +141,70 @@ public IEnumerator ContextTestWithAdditionalWait()
+            bool exitOnFailure = false;
+            int count = 0;
+            bool sendNextResult = true;
+            float lastResultSentTime = 0;
             yield return new ExecuteStepInContext(StepExecutionContext.Clients, _ =>
             {
-                int count = 0;
-
                 void UpdateFunc(float _)
                 {
-                    TestCoordinator.Instance.WriteTestResultsServerRpc(count++);
-                    if (count > maxValue)
+                    if (!exitOnFailure)
                     {
-                        NetworkManager.Singleton.gameObject.GetComponent<CallbackComponent>().OnUpdate -= UpdateFunc;
+                        // We will send if we have another result to send =or= if we haven't received a response back from the server yet.
+                        if (sendNextResult || lastResultSentTime < Time.realtimeSinceStartup)
+                        {
+                            if (!sendNextResult && lastResultSentTime < Time.realtimeSinceStartup)
+                            {
+                                Debug.LogWarning($"{nameof(ContextTestWithAdditionalWait)} is resending due to timeout receiving result received response for count value = {count}!");
+                            }
+
+                            TestCoordinator.Instance.WriteTestResultsServerRpc(count);
+                            lastResultSentTime = resendResultTime + Time.realtimeSinceStartup;
+                            sendNextResult = false;
+                        }
+
+                        // if we exceeded our maximum count, then remove ourselves from the update loop
+                        if (count >= maxValue)
+                        {
+                            NetworkManager.Singleton.gameObject.GetComponent<CallbackComponent>().OnUpdate -= UpdateFunc;
+                            TestCoordinator.Instance.OnServerReceivedResultsResponse = null;
+                        }
                     }
                 }
 
+                void TestResultReceived(float result)
+                {
+                    if (result != count)
+                    {
+                        TestCoordinator.Instance.WriteErrorServerRpc($"Received result {result} but was expected {count} for test {nameof(ContextTestWithAdditionalWait)}!");
+                        NetworkManager.Singleton.gameObject.GetComponent<CallbackComponent>().OnUpdate -= UpdateFunc;
+                        TestCoordinator.Instance.OnServerReceivedResultsResponse = null;
+                        exitOnFailure = true;
+                    }
+                    else
+                    {
+                        count++;
+                        sendNextResult = true;
+                    }
+                }
                 NetworkManager.Singleton.gameObject.GetComponent<CallbackComponent>().OnUpdate += UpdateFunc;
+                TestCoordinator.Instance.OnServerReceivedResultsResponse = TestResultReceived;
```

Comment:
mmm I was about to tell you about yield return new WaitUntil(TestCoordinator.ResultIsSet());
but I'm remembering you can't yield in ExecuteStepInContext... that's a pain. And I think it could be doable to add this (however it'd be way out of scope of your current work :P )

Reply:
there's already a way to do that though. That's what we're doing below server side with                         
Assert.That(res, Is.EqualTo(current++));

Reply:
I'd say, let's remove this if it's not needed. Do we think checking if data sent is data received in a subsequent step is enough?

Reply:
@SamuelBellomo and @NoelStephensUnity is this still an open point of discussion? How do we close this?

Reply:
As agreed in our offline discussion, this should be removed

Reply:
Yes. we agreed that this should be removed too.